celestia.go

```diff
@@ -110,14 +136,29 @@ func (c *CelestiaDA) Validate(ids []da.ID, daProofs []da.Proof) ([]bool, error)
 		proofs = append(proofs, proof)
 	}
 	for i, id := range ids {
-		// TODO: extract commitment from ID
-		isIncluded, err := c.client.Blob.Included(c.ctx, c.height, share.Namespace(c.namespace), proofs[i], blob.Commitment(id))
-		if err != nil {
-			return nil, err
-		}
+		height, commitment := splitID(id)
+		// TODO(tzdybal): for some reason, if proof doesn't match commitment, API returns (false, "blob: invalid proof")
+		//    but analysis of the code in celestia-node implies this should never happen - maybe it's caused by openrpc?
+		//    there is no way of gently handling errors here, but returned value is fine for us
+		isIncluded, err := c.client.Blob.Included(c.ctx, height, c.namespace, proofs[i], commitment)
+		log.Println("Included: ", isIncluded, err)
 		included = append(included, isIncluded)
 	}
 	return included, nil
 }
 
+func makeID(height uint64, commitment da.Commitment) da.ID {
+	id := make([]byte, 8+len(commitment))
```

Comment:
For better code hygiene, instead of just using the number 8 directly, I suggest making a constant equal to `binary.Size(uint64(0))` and using that instead of the number 8.

Reply:
I extracted a `const`, but with literal `8` and comment, as `binary.Size(uint64(0))` is not a constant expression.